### PR TITLE
мерж

### PR DESCRIPTION
**Pull request by OdSt**

### CODE REVIEW

File: code/WorkInProgress/animusstation/optics/beam.dm [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+	HasEntered(var/atom/movable/AM)
+		if(istype(AM, /obj/beam))
+			return
+		if(blocked(AM))
+			remove(src)
+			if(prev)
+				prev.propagate()
+			else if(master)
+				master:turn_on()
```
Comment (Iamgoofball) [inline code/WorkInProgress/animusstation/optics/beam.dm:108]: Colons in general are evil things that can cause runtimes. If using a . won't compile, use typecasting to bypass it.

File: code/__HELPERS/text.dm [hunk 2]
```diff
@@ -43,9 +57,19 @@
 			index = findtext(t, char)
 	return t
 
+proc/sanitize_russian(var/msg) //���������� ��� �����, ��� �� ����� ������� �������� ����� � ������.
```
Comment (Iamgoofball) [inline code/__HELPERS/text.dm:60]: Grabbing some of this stuff could be nice if we get any surprise visits from Russian SS13 players accidentally joining the english servers.

File: code/_onclick/hud/screen_objects.dm [hunk 3]
```diff
@@ -224,7 +228,20 @@
 						else
 							C << "<span class='notice'>You don't have an internals tank!</span>"
 		if("act_intent")
-			usr.a_intent_change("right")
+			switch(icon_y)
```
Comment (Iamgoofball) [inline code/_onclick/hud/screen_objects.dm:231]: This is one of my favourite features about bay that we should really grab sometime.

File: code/modules/mob/living/carbon/human/emote.dm [hunk 4]
```diff
@@ -278,6 +285,44 @@
...
+		if("elaugh")
+			if(miming)
+				message = "<B>[src]</B> acts out a laugh."
+				m_type = 1
+			else if (mind.special_role)
+				if (!ready_to_elaugh())
+					if (world.time % 3)
+						usr << "<span class='warning'>You not ready to laugh again!"
+				else
+					message = "<B>[src]</B> laugh like a true evil! Mu-ha-ha!"
```
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/human/emote.dm:310]: > laugh like a true evil

File: code/modules/powerarmor/components/armor.dm [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+	else if(istype(W, /obj/item/stack))
+		var/obj/item/stack/S = W
+
+		if(istype(S, /obj/item/stack/sheet/plasteel) && upgrades["plasteel"] < 4)
+			upgrades["plasteel"]++
+			add_armor(list(melee = 4, bullet = 8, energy = 2, bomb = 8, rad = 2))
+			user << "<span class='info'>You strengthen [src], improving its resistance against bullets and bombs.</span>"
+			S.use(1)
+
+		else if(istype(S, /obj/item/stack/sheet/xenochitin) && upgrades["xenochitin"] < 4)
```
Comment (Iamgoofball) [inline code/modules/powerarmor/components/armor.dm:83]: Finally, someone found a use for this shit.

File: code/modules/powerarmor/suit.dm [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+	else if(!active)
+		if(istype(W, /obj/item/weapon/crowbar))
+			if(subsystems.len)
+				for(var/obj/item/weapon/powerarmor/I in subsystems)
+					I.drop()
+				user.visible_message("<span class='warning'>[user] removes [src]'s subsystems!</span>",
+				"<span class='warning'>You remove [src]'s subsystems!</span>",
+				"<span class='notice'>You hear a click!</span>")
+
+			else if(meele)
```
Comment (Iamgoofball) [inline code/modules/powerarmor/suit.dm:243]: melee is the proper spelling

it's like this all over the file

File: code/modules/reagents/reagent_containers/food/drinks.dm [hunk 7]
```diff
@@ -165,6 +165,18 @@
 		src.pixel_x = rand(-10.0, 10)
 		src.pixel_y = rand(-10.0, 10)
 
+/obj/item/weapon/reagent_containers/food/drinks/flour/attackby(obj/item/weapon/W as obj, mob/user as mob)
```
Comment (Iamgoofball) [inline code/modules/reagents/reagent_containers/food/drinks.dm:168]: This is a pretty nice way to make dough, but did you know you normally make the dough by putting the flour container on the table, putting an egg on it, and then dragging the table sprite to your guy?

File: code/modules/research/designs.dm [hunk 8]
```diff
@@ -1595,7 +1595,7 @@ datum/design/lasercannon
 	build_type = PROTOLATHE
 	materials = list("$metal" = 10000, "$glass" = 10000, "$diamond" = 2000)
 	build_path = /obj/item/weapon/gun/energy/lasercannon
-	locked = 1
```
Comment (Iamgoofball) [inline code/modules/research/designs.dm:1598]: :+1: x :100:

**Comment by Atskadan** [comment]
what

**Comment by Iamgoofball** [comment]
basically what's going on seems to be:

Russian SS13 servers are making the switch to Github

They forked NT and Bay, and other codebases.

They try to make PRs to their forks, but accidentally target them to our fork

This has been going on for awhile at BS12,

**Pull request closed without merging**